### PR TITLE
Learn page glowup

### PR DESCRIPTION
**Pull request by napkinbrown**

### CODE REVIEW

File: learn.md [hunk 1]
```diff
@@ -1,50 +1,262 @@
 ---
-layout: learn
+layout: page
 title: Learn
 permalink: /learn/
+toc: true
```
Comment (napkinbrown) [inline learn.md:5]: delete this

File: learn.md [hunk 2]
```diff
@@ -1,50 +1,262 @@
...
+## Tools:
+
+### Required Tools to start for free today
+
+- [Ghidra](https://ghidra-sre.org/)
+  - Use this to decompile binaries and executables
+- [GDB](gnu.org/software/gdb/)
+  - Useful for dynamic analysis
+- [Hex Editor](https://hexed.it/)
+  - Useful for editing binaries
```
Comment (napkinbrown) [inline learn.md:144]: Can you add examples of basic use cases for each of these tools? See the Burp section for kind of what I'm looking for. If I installed Ghirda, what's one thing I can use immediately for pretty much anything? (previewing the disassembled C code comes to mind). 

Obviously you don't (and shouldn't) need to cover every scenario. If the tool has a lot of varied use, link to external tutorials or guides. 

Screenshots and code blocks are encouraged

File: learn.md [hunk 3]
```diff
@@ -1,50 +1,262 @@
...
+- [MobSF](https://mobsf.github.io/docs/#/)
+  - All in one APK tool 
+  - Extracts all APK data
+  - Decompiles all java
+  - Sets up emulation for dynamic analysis
+#### Compiled python challenges
+- [Uncompyle](https://pypi.org/project/uncompyle6/)
+    - Decompiles compiled python bytecode
+- [Pyinstxtractor](https://github.com/extremecoders-re/pyinstxtractor)
+  - Used for extracting compiled python from executables and binaries made with pyinstaller
```
Comment (napkinbrown) [inline learn.md:183]: 👍

File: learn.md [hunk 4]
```diff
@@ -1,50 +1,262 @@
...
+
+## How to Learn:
+
+### Practice, Practice, Practice. 
+- Reverse engineering is extremely time consuming, especially when you start, but as you practice you will start recognizing pattern and get faster each time.
+- Almost all good Reverse engineering challenges will be unique and will require that you will need to learn about something new, this skill improves as you practice.
+- You can practice lots of challenges [here](https://github.com/Kasimir123/CTFWriteUps). All the challenges are rated for difficulty and each challenge has the original files and a writeup.
```
Comment (napkinbrown) [inline learn.md:190]: Move this above the "Tools" section

File: learn.md [hunk 5]
```diff
@@ -1,50 +1,262 @@
...
+
+## Must-Have Tools
+- [Ghidra](https://ghidra-sre.org/) - for light reverse engineering to find vulnerability
+- [GDB](http://www.unknownroad.com/rtfm/gdbtut/gdbtoc.html) - for tracing how the program handles your input
+- [Pwntools](https://github.com/Gallopsled/pwntools) - A Python library for Pwn solution scripts. Useful for sending/receiving input, reading memory, and more.
+
+
+
+# Crypto
+
```
Comment (napkinbrown) [inline learn.md:221]: This should be swapped out with the guide written by the astrophysicist 😄

File: learn.md [hunk 6]
```diff
@@ -1,50 +1,262 @@
...
+### Practice, Practice, Practice. 
+- Reverse engineering is extremely time consuming, especially when you start, but as you practice you will start recognizing pattern and get faster each time.
+- Almost all good Reverse engineering challenges will be unique and will require that you will need to learn about something new, this skill improves as you practice.
+- You can practice lots of challenges [here](https://github.com/Kasimir123/CTFWriteUps). All the challenges are rated for difficulty and each challenge has the original files and a writeup.
+
+
+### GuidedHacking is an amazing forum with a lot of information about game hacking.
+- Always remember to only hack things you have permission to hack/ try to stay on single player games to stay safe.
+- Following their tutorials and hacking various games you get exposure to multiple architectures, learn reverse engineering tricks, and you will learn about how to get around anti debug code.
```
Comment (napkinbrown) [inline learn.md:195]: Maybe move this to an "Other Topics" section? Game Hacking isn't very relevant to intro CTF reverse engineering

File: learn.md [hunk 7]
```diff
@@ -1,50 +1,262 @@
...
+
+# Binary Exploitation (Pwn)
+
+Binary Exploitation (pwn) challenges involve finding and exploiting vulnerabilities in executable programs, typically to gain a remote shell. Pwn challenges tend to have a higher learning curve than the other categories. A familiar understanding of Linux, C, assembly, are recommended before doing pwn challenges.
```
Comment (napkinbrown) [inline learn.md:199]: "Well crap, I don't know Linux, C, or Assembly. What should I do?"

File: learn.md [hunk 8]
```diff
@@ -1,50 +1,262 @@
...
+
+## Learning Resources
+- Learn Assembly and C
+  - [Comparing C to Machine Language](https://www.youtube.com/watch?v=yOyaJXpAYZQ)
+- Learn about the stack and calling conventions
+- [LiveOverflow's Binary Exploitation Playlist](https://www.youtube.com/watch?v=iyAyN3GFM7A&list=PLhixgUqwRTjxglIswKp9mpkfPNfHkzyeN) - This playlist is an incredible resource for learning binary exploitation, and will walk you through nearly all techniques used in entry-mid level pwn challenges
```
Comment (napkinbrown) [inline learn.md:205]: Consider embedding the first video/entire playlist

File: learn.md [hunk 9]
```diff
@@ -217,46 +271,25 @@ Binary Exploitation (pwn) challenges involve finding and exploiting vulnerabilit
 
 
 
-# Crypto
-
-## Basic Ciphers
-
-### Cesear Cipher
```
Comment (weakphish) [inline learn.md]: should be Caesar

File: learn.md [hunk 10]
```diff
@@ -217,46 +271,25 @@ Binary Exploitation (pwn) challenges involve finding and exploiting vulnerabilit
...
-
-_What?_
-
-[Ceaser Cipher](https://en.wikipedia.org/wiki/Caesar_cipher) is a simple substitution cipher where a specific number (the key) shifts the alphabet. For example, if the key were 3, A would turn into C, B into D (and so on). If the key was 26 (or 0) A would simply map to A.
```
Comment (weakphish) [inline learn.md]: same here

**Review by weakphish — COMMENTED** [review]
lgtm

**Pull request merged**